### PR TITLE
Disabled compiling warnings in case of symbols in cmake for 3.4

### PR DESCRIPTION
**Pull request by asenyaev**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [ ] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

```
force_builders=Win64
build_image:Custom Win=msvs2019
```

### CODE REVIEW

File: cmake/OpenCVCompilerOptions.cmake [hunk 1]
```diff
@@ -426,6 +426,8 @@ if(MSVC)
     ocv_warnings_disable(CMAKE_CXX_FLAGS /wd4275) # non dll-interface class 'std::exception' used as base for dll-interface class 'cv::Exception'
     ocv_warnings_disable(CMAKE_CXX_FLAGS /wd4512) # Assignment operator could not be generated
     ocv_warnings_disable(CMAKE_CXX_FLAGS /wd4589) # Constructor of abstract class 'cv::ORB' ignores initializer for virtual base class 'cv::Algorithm'
+    ocv_warnings_disable(CMAKE_CXX_FLAGS /wd4819) # Symbols like delta or epsilon cannot be represented
+    ocv_warnings_disable(CMAKE_CXX_FLAGS /wd4702) # unreachable code using Ninja
```
Comment (alalek) [inline cmake/OpenCVCompilerOptions.cmake:430]: > unreachable code

We definitely should not disable such warning globally.

Local `#pragma` should be used instead.

Comment (asenyaev) [reply cmake/OpenCVCompilerOptions.cmake]: Removed this line from [cmake/OpenCVCompilerOptions.cmake](https://github.com/opencv/opencv/pull/22285/files/3e12ebb7aa335c3ce686ab742681f124c7ebb631#diff-1cfc4f7f7baf9f200210e211eedcc4a44a494a3bc3ef887f328aac2f0716b39b) and added `#pragma` to certain files.

File: modules/core/include/opencv2/core/matx.hpp [hunk 2]
```diff
@@ -57,6 +57,11 @@
 #include <initializer_list>
 #endif
 
+// WARNING: unreachable code using Ninja
+#if defined _MSC_VER && _MSC_VER >= 1928
+#pragma warning(disable: 4702)
+#endif
+
```
Comment (alalek) [inline modules/core/include/opencv2/core/matx.hpp:64]: Disabling in such way in a public header is a way similar to global disabling.

Could you provide link on build logs with these warnings?

Comment (asenyaev) [reply modules/core/include/opencv2/core/matx.hpp:64]: Yes, sure: https://github.com/opencv/opencv/runs/7454965078?check_suite_focus=true#step:13:584

Do you mean to disable each line with this warning?

Comment (asenyaev) [reply modules/core/include/opencv2/core/matx.hpp]: Removed the global definition and put to certain places which happen during the build.

**Comment by asmorkalov** [comment]
@alalek May we merge this?

File: modules/core/include/opencv2/core/cuda.hpp [hunk 3]
```diff
@@ -488,8 +488,16 @@ class CV_EXPORTS BufferPool
     //! Allocates a new GpuMat of given size and type.
     GpuMat getBuffer(int rows, int cols, int type);
 
+// WARNING: unreachable code using Ninja
+#if defined _MSC_VER && _MSC_VER >= 1928
```
Comment (alalek) [inline modules/core/include/opencv2/core/cuda.hpp:492]: > 1928

Just 1920 (MSVS 2019 base version) as we probably have we same problem with 1927 etc.

https://learn.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170

Comment (asenyaev) [reply modules/core/include/opencv2/core/cuda.hpp]: I just wanted to pin a version which caught a warning and I didn't test it on 1920, but updated changes according to your comment.

**Review by alalek — APPROVED** [review]

**Review by asmorkalov — APPROVED** [review]

**Pull request merged**